packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -2839,6 +2852,10 @@ def __eq__(self, other: Any) -> bool:
             return self.to_dict() == other.to_dict()
         return False
 
+    def has_bad_availability_rate(self, threshold: float = 0.95) -> bool:
+        """Whether the agent on which the window belongs to has a bad availability rate or not."""
+        return self._num_positive >= ceil(self._max_length * threshold)
```

Comment:
maybe it makes sense to have some minimum max_length, to avoid slashing in the begging.

Reply:
You are right, though the confusion stands on the fact that I marked the offensive behaviours as positive and the normal as negative, even though it is called an "availability" window. We could swap this in the future because I agree it is confusing. The explanation in the code:

https://github.com/valory-xyz/open-autonomy/blob/0ef9a35d70047f7aa9b31ae0d19e25d3e0e20d0d/packages/valory/skills/abstract_round_abci/base.py#L3420

we add in case the agent was down, i.e., increase the positives:

https://github.com/valory-xyz/open-autonomy/blob/0ef9a35d70047f7aa9b31ae0d19e25d3e0e20d0d/packages/valory/skills/abstract_round_abci/base.py#L2903

&&

https://github.com/valory-xyz/open-autonomy/blob/0ef9a35d70047f7aa9b31ae0d19e25d3e0e20d0d/packages/valory/skills/abstract_round_abci/base.py#L2884